booking-deeplink-specification.md

```diff
@@ -0,0 +1,24 @@
+# Indroduction
+
+This specification provides precise guidelines for a carpooling operator and a MaaS platform to implement a "booking" flow for passengers, with information to complete the booking shared via deeplinking between the two.
+
+## Requirements Notation and Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "PROVIDERTIONAL" in this document are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
+
+## Functional overview
+
+The booking flow with a deep link starts when a passenger clicks on a journey search result in the MaaS app. This search result MUST open the deep link contained in the [DeepLink](https://github.com/klaxit/open-carpooling-service-specification/blob/c34cbfdea9991bfd3453d0d0822066411dcfa219/carpool-api-en.yml#L268) parameter returned by the [CarpoolJourney](https://github.com/klaxit/open-carpooling-service-specification/blob/c34cbfdea9991bfd3453d0d0822066411dcfa219/carpool-api-en.yml#L140) object in the [carpoolJourneys](https://github.com/klaxit/open-carpooling-service-specification/blob/master/carpool-api-en.yml#L10) response. This parameter MUST be provided by the carpool operator responsible for their implementation of the search API public specification. It contains the full URL and its corresponding parameters needed for the mobile device to redirect the passenger from the app displaying the search results to the operator's app.
+
+One obstacle to redirecting the passenger is whether they have the operator's app already installed in their mobile device or not. If not, the deep link URL SHOULD be capable of redirecting the passenger to the corresponding store instead. At the operator's discretion, this URL can also temporarily store the URL parameters for later retrieval. There are third-party services as well that can provide such a solution. This way, after the app is installed, the parameters can be recovered to automatically redirect the passenger from the onboarding to the booking flow as if the app had always been installed.
+
+If the app was already installed and the passenger was already registered with the operator service, then clicking on the deep link will automatically and seamlessly redirect the passenger to the operator's booking flow. It is up to the operator to decide where to start this flow and what to do with the parameters sent to the MaaS app in the search to provide the passenger with the best booking experience. It is up to the passenger to decide whether to book the clicked journey or not.
```

Comment:
"[...] then clicking on the deep link SHOULD automatically and seamlessly [....]"